core/modules/image/image.field.inc

```diff
@@ -315,13 +315,17 @@ function image_field_presave($entity_type, $entity, $field, $instance, $langcode
 
   // Determine the dimensions if necessary.
   foreach ($items as &$item) {
-    if (!isset($item['width']) || !isset($item['height'])) {
+    if (!isset($item['width']) || !isset($item['height']) || empty($item['width']) || empty($item['height'])) {
       $info = image_get_info(file_load($item['fid'])->uri);
 
       if (is_array($info)) {
         $item['width'] = $info['width'];
         $item['height'] = $info['height'];
       }
+      else {
```

Comment:
At this point it is clear that this file is severely broken. It might be smarter to not handle it as image anymore.